.circleci/config.yml

```diff
@@ -680,6 +624,56 @@ jobs:
           command: |
             gh-pages --dotfiles --message "[skip ci] documentation update" --dist target/doc
 
+  ######################################################################################################
+  # Test building docker ci base image if relevnat files have changed                                  #
+  ######################################################################################################
+  docker-ci-build:
+    executor: docker-executor
+    description: publish docker images
+    steps:
+      - checkout
+      - setup_docker_signing
+      - run:
+          name: Build and publish the ci base image.
+          command: |
+
+            #detect if any relevant files have changed requiring base docker file rebuilding.
+            #this is specific for prs targeting master.   TODO: figure out target release branch.
+            git --no-pager diff --name-only HEAD $(git merge-base HEAD origin/master) | sort > /tmp/changed_files
```

Comment:
What if the PR doesn't squash and have multiple commits?  May need to check changed files in  `git rev-list orgin/master..HEAD`.

Reply:
I don't care about intermediate changes, just all changes between the current head and the merge base with master (where the branch was cut).   This will handle the case you mention.